src/lib/Libdb/db_postgres_attr.c

```diff
@@ -237,37 +239,28 @@ dbarray_to_attrlist(char *raw_array, pbs_db_attr_list_t *attr_list)
  *
  * @return      Error code
  * @retval	-1 - On Error
- * @retval	 0 - On Success
+ * @retval	 length of array - On Success
  *
  */
 int
 attrlist_to_dbarray_ex(char **raw_array, pbs_db_attr_list_t *attr_list, int keys_only)
 {
-	struct pg_array *array;
-	int len = 0;
+	/* use static variables to improve performance by not allocating memory for each object save */
+	static struct pg_array *array = NULL, *tmp;
+	static int len = sizeof(struct pg_array) + DBARRAY_BUF_LEN;
 	struct str_data *val = NULL;
-	int attr_val_len = 0;
 	svrattrl *pal;
-	int count = 0;
-
-	len = sizeof(struct pg_array);
-	for (pal = (svrattrl *)GET_NEXT(attr_list->attrs); pal != NULL; pal = (svrattrl *)GET_NEXT(pal->al_link)) {
-		len += sizeof(int32_t) + PBS_MAXATTRNAME + PBS_MAXATTRRESC + 1; /* include space for dot */
-		if (keys_only)
-			attr_val_len = 0;
-		else
-			attr_val_len = (pal->al_atopl.value == NULL ? 0 : strlen(pal->al_atopl.value));
-		len += sizeof(int32_t) + 3 + attr_val_len + 1; /* include space for dot */
-		count++;
+	char *p;
+	int spc_avl, spc_req;
+	/* len attr_name . attr_resc len attr_flags . attr_value */ 
+	static int fixed_part_req = sizeof(int32_t)*2 + PBS_MAXATTRNAME + PBS_MAXATTRRESC + 3 /* max 3 digits flags */ + 2 /* 2 dots */ + 1 /* null terminator */; 
```

Comment:
space around '*' is needed.
Also, having comments within the statement is not much readable.

Reply:
will put space, but not sure how to put comment to match each piece, let me try